week2/ShoppingList/cart.js

```diff
@@ -0,0 +1,289 @@
+const ITEM_LIST_KEY = "itemList";
+//장바구니(cart)페이지 구현 - 로컬 스토리지 활용
+const cartBody = document.getElementById('cartBody');
+
+const storageItems = localStorage.getItem(ITEM_LIST_KEY);
+const parsedItems = JSON.parse(storageItems); //localStorage에서 가져온 아이템
+
+function deleteGrandParent(e)
+{
+  const grandParent = (e.currentTarget.parentNode).parentNode;
+  //해당 노드의 id를 확인하여, 
+  //로컬 스토리지에 저장된 객체 배열들의 요소들 중,
+  //같은 id를 갖는 객체를 cart = false; 설정해주기.
+  parsedItems[grandParent.id -1].cart = false;
+  parsedItems[grandParent.id -1].userCart = false; //처음엔 이거 안해줬는데, 삭제 후 추가하니 자동 체크되는 상황 발생하여 수정함
+  //그 후, 다시 로컬스토리지도 갱신
+  localStorage.setItem(ITEM_LIST_KEY, JSON.stringify(parsedItems));
+
+  grandParent.remove(); //조부모 삭제
+}
+
+/* 로컬 스토리지에서 값을 가져와서, 체크 박스 체크 해주는 함수*/
+function initSelect()
+{
+  console.log("init실행");
+  const initParsedItems = JSON.parse(localStorage.getItem(ITEM_LIST_KEY));
+  //로컬 스토리지 기록으로 체크 해줄 checkbox들 전부 가져옴
+  const checkboxes = document.querySelectorAll("#cartBody input[type ='checkbox']") //cartbody에 있는 모든 checkbox 타입의 input을 배열로 가져옴
+  checkboxes.forEach((box)=>{
+    const grandNode = box.parentElement.parentElement;
+    if(initParsedItems[grandNode.id -1].userCart === true) //userCart 즉, 체크여부
+    {
+      box.checked = true; //이건 그냥 check로만 만들어주는 것! (그래서 신경쓸 게 생각보다 없음)
+    }
+  });
+}
+
+
+
+
+
+function selected(e)
+{
+  const grandParent = (e.currentTarget.parentNode).parentNode;
+  if(e.currentTarget.checked) //체크 되어있는지 확인(boolean값임)
+  {
+    parsedItems[grandParent.id -1].userCart = true;
+  }
+  else{ //체크 안되어 있으면(혹은 체크 해제했으면)
+    parsedItems[grandParent.id -1].userCart = false;
+  }
+  //로컬 스토리지에 갱신
+  localStorage.setItem(ITEM_LIST_KEY, JSON.stringify(parsedItems));
+}
+
+function paintCart(obj)
+{
+  if(obj.cart === true)
+  {
+    const tr = document.createElement("tr");
+    tr.id = obj.id; //나중에 삭제, 구매 확정에 사용될 예정
+
+    // 첫번째 열(체크버튼) 생성 후, tr안에 넣어두는 과정
+    const td1 = document.createElement("td");
```

Comment:
열이 순서가 있어서 숫자를 쓰는 방법도 좋지만 저는 개인적으로 같은 변수 이름에 숫자만 바꾸면 나중에 헷갈리더라고요... 🥺
```suggestion
    const tdCheckbox = document.createElement("td");
```
이런 식으로 바꾸는 건 어떨까요?? 취향 차이인 것 같아서 한 번 고민해 보시면 좋을 것 같아요! 🔥

Reply:
좋은 의견인 것 같아요!! 코드에 반영해보도록 하겠습니다 😊